### PR TITLE
move prepayment package from fos to foo

### PR DESCRIPTION
**Pull request by julianzimmermann**

### CODE REVIEW

File: bundles/prepayment/composer.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  "require": {
+    "php": ">=7.2",
+    "spryker/checkout": "^6.2.1",
+    "spryker/kernel": "^3.42.4",
+    "spryker/oms": "^11.1.0",
+    "spryker/payment": "^5.0.0",
+    "spryker/refund": "^5.3.0",
+    "spryker/sales": "^11.4.1",
+    "spryker/step-engine": "^3.2.2",
+    "spryker/symfony": "^3.3.5"
```
Comment (daniel-rose) [inline bundles/prepayment/composer.json:20]: Please check versions...

File: bundles/prepayment/composer.json [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  "description": "n/a",
+  "license": "MIT",
+  "authors": [
+    {
+      "name": "Daniel Rose",
+      "email": "daniel-rose@gmx.de"
+    }
+  ],
+  "require": {
+    "php": ">=7.2",
```
Comment (daniel-rose) [inline bundles/prepayment/composer.json:12]: EOL for php 7.2. Use >= 7.3.

File: bundles/prepayment/composer.json (hunk 1 above)
Comment (daniel-rose) [reply bundles/prepayment/composer.json:20]: Please check required packages. Kernel is not required for example...

**Review by daniel-rose — CHANGES_REQUESTED** [review]
Please add unit tests...

File: bundles/prepayment/codeception.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+namespace: FondOfOryx
```
Comment (daniel-rose) [inline bundles/prepayment/codeception.yml:1]: wrong namespace

File: bundles/prepayment/src/FondOfOryx/Yves/Prepayment/Form/DataProvider/PrepaymentFormDataProvider.php [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+     * @param \Spryker\Shared\Kernel\Transfer\AbstractTransfer $dataTransfer
+     *
+     * @return \Spryker\Shared\Kernel\Transfer\AbstractTransfer
+     */
+    public function getData(AbstractTransfer $dataTransfer)
+    {
+        if (method_exists($dataTransfer, 'getPayment') && method_exists($dataTransfer, 'setPayment') && $dataTransfer->getPayment() === null) {
+            $paymentTransfer = new PaymentTransfer();
+
+            $dataTransfer->setPayment($paymentTransfer);
```
Comment (daniel-rose) [inline bundles/prepayment/src/FondOfOryx/Yves/Prepayment/Form/DataProvider/PrepaymentFormDataProvider.php:21]: Use one line...

File: bundles/prepayment/src/FondOfOryx/Zed/Prepayment/Communication/Plugin/Checkout/PrepaymentPostCheckPlugin.php [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+     *
+     * @return \Generated\Shared\Transfer\CheckoutResponseTransfer
+     */
+    public function execute(QuoteTransfer $quoteTransfer, CheckoutResponseTransfer $checkoutResponseTransfer)
+    {
+        if (!$this->isAuthorizationApproved($quoteTransfer)) {
+            $checkoutErrorTransfer = new CheckoutErrorTransfer();
+            $checkoutErrorTransfer
+                ->setErrorCode(static::ERROR_CODE_PAYMENT_FAILED)
+                ->setMessage('Something went wrong with your payment. Try again!');
```
Comment (daniel-rose) [inline bundles/prepayment/src/FondOfOryx/Zed/Prepayment/Communication/Plugin/Checkout/PrepaymentPostCheckPlugin.php:33]: Use one line...

```php
$checkoutErrorTransfer = (new CheckoutErrorTransfer())
    ->setErrorCode(static::ERROR_CODE_PAYMENT_FAILED)
    ->setMessage('Something went wrong with your payment. Try again!');
```

File: bundles/prepayment/src/FondOfOryx/Zed/Prepayment/PrepaymentConfig.php [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+use Spryker\Zed\Kernel\AbstractBundleConfig;
+
+class PrepaymentConfig extends AbstractBundleConfig
+{
+    /**
+     * @return string|null
+     */
+    public function getIban(): ?string
+    {
+        return $this->get(PrepaymentConstants::IBAN);
```
Comment (daniel-rose) [inline bundles/prepayment/src/FondOfOryx/Zed/Prepayment/PrepaymentConfig.php:15]: Are you sure, that null is allowed?

**Review by daniel-rose — APPROVED** [review]

**Review by gengjozsef — APPROVED** [review]

**Pull request merged**